### PR TITLE
feature: tweak documentation structure

### PR DESCRIPTION
**Pull request by bitpshr**
This pull request updates the structure of the documentation site in a few ways:

- The "Getting Started" page was expanded into a "Getting Started" section since there's a lot of content for one page now (especially if more client configuration parameters are ever added.)
- `Abstract.Sketch` API docs were moved into the "Reference" section alongside `Abstract.Client` docs so all API docs are in one place.
- A new "Concepts" section was added for things like pagination (#32), requesting latest commits ([#40](https://github.com/goabstract/abstract-sdk/issues/40#issuecomment-443562588)), and other concepts we identify as important-enough to document independently.

Resolves https://github.com/goabstract/abstract-sdk/issues/44

_Overview of modified structure_

installation

_New "Request Latest Commit" page_

request-latest

### CODE REVIEW

File: docs/authentication.md [hunk 1]
```diff
@@ -7,6 +7,6 @@ Authentication with the SDK is achieved by using an OAuth access token. During t
 
   > API tokens should be treated like a password, as they provide access to all data that your user has access to. Do not commit them to version control.
 
-## Usage
+## Using a token
 
-To use the token simply save it in your environment as `ABSTRACT_TOKEN` and it will automatically be used. You can also use another method to load the token and pass it to the SDK on initialization – see [Getting Started](/docs/getting-started) for more details.
+To use the token simply save it in your environment as `ABSTRACT_TOKEN` and it will automatically be used. You can also use another method to load the token and pass it to the SDK on initialization – see [Configuration](/docs/configurationi) for more details.
```
Comment (tommoor) [inline docs/authentication.md]: The link is bad here "configurationi"

**Comment by tommoor** [comment]
Looks great in general, my only concern with this change is that having the bulk of the API documentation kind of hidden underneath "Abstract.Client" isn't very clear to me…

File: docs/configuration.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+---
+id: configuration
+title: Configuration
+---
+
+In order to use the SDK you'll need to instantiate an instance and give it an [authentication](/docs/authentication) token. We recommend storing the token in your environment as `ABSTRACT_TOKEN` and it will be automatically loaded – don't commit it with your code!
```
Comment (amccloud) [inline docs/configuration.md]: `need to instantiate a client`

File: docs/configuration.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+## Transports
+
+If you want to ensure that the SDK only ever loads data from the API or the CLI then you can achieve this by specifying a transport. This is useful when running in an environment without the Mac application installed or alternatively when you want to ensure you're only dealing with local data:
+
+```js
+import * as Abstract from "abstract-sdk";
+
+const abstract = Abstract.Client({
+  transport: Abstract.TRANSPORTS.CLI
```
Comment (amccloud) [inline docs/configuration.md]: suggestion:

```
  transport: Abstract.TRANSPORTS.CLI // or Abstract.TRANSPORTS.API
```

File: docs/latest.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+---
+
+When using the SDK to query for various types of resources, it's common to pass in an `sha` property to limit results to a specific commit on a given project branch. In addition to specifying an actual SHA hash, the latest commit can always be targeted by passing the string `"latest"`.
+
+For example, it's possible to fetch activities for the latest commit on the master branch of a project without knowing the actual commit hash:
+
+```js
+abstract.activities.list({
+  branchId: "master",
+  commitId: "latest",
```
Comment (amccloud) [inline docs/latest.md]: sha

File: docs/latest.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
+---
+id: latest
+title: Request latest commit
+---
+
+When using the SDK to query for various types of resources, it's common to pass in an `sha` property to limit results to a specific commit on a given project branch. In addition to specifying an actual SHA hash, the latest commit can always be targeted by passing the string `"latest"`.
+
+For example, it's possible to fetch activities for the latest commit on the master branch of a project without knowing the actual commit hash:
```
Comment (amccloud) [inline docs/latest.md]: I'm not sure latest will actually work on activities. As of now `latest` as the sha is only supported on `LayerDescriptor` and `FileDescriptor` and can only be used where those descriptors can be used

**Comment by bitpshr** [comment]
Thanks for the feedback so far @amccloud and @tommoor, this should be ready for another pass.

Re: the main documentation being buried under `Abstract.Client`, I'm torn: I agree that it's not very obvious, but it's accurate per the structure of the API, which I do think is important if we're labeling it as actual API documentation. Maybe we can ditch the `Abstract.` package namespace in each menu item since it's sort of redundant, and move `Sketch` to a `util` namespace?

**Comment by amccloud** [comment]
@bitpshr 👍 on removing the prefix. Could maybe even have Client and Sketch top-level

**Comment by tommoor** [comment]
> move Sketch to a util namespace

Could you say more about this?

**Comment by bitpshr** [comment]
@tommoor thinking ahead, will we have additional utility modules beyond `Sketch`? If so, it may make sense to nest those modules in a `util` namespace for clarity and organization:

```js
import {
    Client,
    util
} from "abstract-sdk";

const abstract = Client();

util.Sketch.isSketchPlugin();
util.Illustrator.someMethod();
```

Totally secondary to downstream use, this would also alleviate the documentation clarity issue in this PR since the API menu would have `Client` at the top level with `Sketch` nested under `util`, so we could one of the following:

nested

Or...

inline

**Comment by tommoor** [comment]
Gotcha. I'd vote to keep it top level for now, if we end up with a top-level namespace for each tool we support that's not necessarily a bad thing – util / lib folders tend to become a dumping ground pretty fast 😅

**Comment by bitpshr** [comment]
Ok, I ditched the `Abstract.` namespace on each menu item and left `Sketch` top level for now. This should be ready for another pass @tommoor and @amccloud.

Playing devil's advocate re: `util` outside of this PR (though I'm definitely not tied to it): without a `util` folder (or something more purposeful, like `vendor`) to place `Sketch` and other product-specific utilities, the root of the SDK package itself becomes the dumping ground. I've never felt that the potential dumping ground argument was strong reason to avoid clear package structures / module hierarchy. In either case, thought should always be put into the public APIs we expose regardless of what level we expose them at. I do think it's a nonissue for now though.

**Comment by tommoor** [comment]
Sounds good, argument heard. Lets keep an eye on it as the SDK grows in scope and keep iterating!

**Pull request merged**